### PR TITLE
Sass 1163

### PR DESCRIPTION
**Pull request by lz1kwn**
### Description
Create fuel benefit for company car radio button EOY page

Add a link to the relevant story in Jira
[SASS-1163](https://jira.tools.tax.service.gov.uk/browse/SASS-1163)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/controllers/employment/CompanyCarFuelBenefitsController.scala [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                      Some(cya.copy(employmentBenefits = cya.employmentBenefits.map(_.copy(
+                        carVanFuelModel = Some(model.copy(carFuelQuestion = Some(true)))
+                      ))))
+                    } else {
+                      Some(cya.copy(employmentBenefits = cya.employmentBenefits.map(_.copy(
+                        carVanFuelModel = Some(model.copy(carFuelQuestion = Some(false), carFuel = None))
+                      ))))
+                    }
+                  case None =>
+                    //                  TODO: Need to potentially update this to make a cya or something
```
Comment (ina-tu) [inline app/controllers/employment/CompanyCarFuelBenefitsController.scala:85]: TODO can be removed

File: it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+      )
+    )
+  }
+
+  ".show" should {
+
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        "Render the 'Did you receive car benefits' page with the correct content with no benefits data so no pre-filling" which {
```
Comment (ina-tu) [inline it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala:155]: test name can be changed to "car fuel benefits page"

File: it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+          h1Check(user.specificExpectedResults.get.expectedH1)
+          textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+          radioButtonCheck(yesText, 1, Some(false))
+          radioButtonCheck(noText, 2, Some(false))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "Render the 'Did you receive car benefits' page with the correct content with cya data and the yes value pre-filled" which {
```
Comment (ina-tu) [inline it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala:182]: same here

File: it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+          h1Check(user.specificExpectedResults.get.expectedH1)
+          textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+          radioButtonCheck(yesText, 1, Some(true))
+          radioButtonCheck(noText, 2, Some(false))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "Redirect the user to the overview page when theres no session data for that user" which {
```
Comment (ina-tu) [inline it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala:209]: test name can be changed to "redirect user to the check your benefits page"

File: it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true, Some(benefits(fullCarVanFuelModel)))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlPost(carFuelBenefitsPage(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          "redirects to the check your details page" in {
```
Comment (ina-tu) [inline it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala:279]: test name can be changed to check your benefits page

File: it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+            result.header("location") shouldBe
+              Some(s"/income-through-software/return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
+            lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+            cyamodel.employment.employmentBenefits.flatMap(_.carVanFuelModel.flatMap(_.carVanFuelQuestion)) shouldBe Some(true)
+            cyamodel.employment.employmentBenefits.flatMap(_.carVanFuelModel.flatMap(_.carFuelQuestion)) shouldBe Some(true)
+          }
+
+        }
+
+        "Redirect to overview page when theres no CYA data" which {
```
Comment (ina-tu) [inline it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala:290]: same here

File: app/controllers/employment/CompanyCarFuelBenefitsController.scala [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+ */
+
+package controllers.employment
+
+import config.{AppConfig, ErrorHandler}
+import controllers.employment.routes.CheckYourBenefitsController
+import controllers.predicates.{AuthorisedAction, InYearAction}
+import forms.YesNoForm
+import models.User
+import models.employment.CarVanFuelModel
```
Comment (ina-tu) [inline app/controllers/employment/CompanyCarFuelBenefitsController.scala:24]: unused import

File: app/controllers/employment/CompanyCarFuelBenefitsController.scala [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+import models.mongo.EmploymentCYAModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.EmploymentSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.employment.{ CompanyCarFuelBenefitsView}
```
Comment (ina-tu) [inline app/controllers/employment/CompanyCarFuelBenefitsController.scala:32]: braces can be removed from import for CompanyCarFuelBenefitsView

File: it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = {
+    Seq(UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+      UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+      UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+      UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY)))
+  }
+
+  object CyaModel {
```
Comment (ina-tu) [inline it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala:141]: `object CyaModel` is not used and can be removed

File: it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+      carFuel = Some(200),
+      vanQuestion = Some(true),
+      van = Some(300),
+      vanFuelQuestion = Some(true),
+      vanFuel = Some(400),
+      mileageQuestion = Some(true),
+      mileage = Some(400)
+    )
+
+  def emptyCarVanFuelModel: CarVanFuelModel =
```
Comment (ina-tu) [inline it/controllers/employment/CompanyCarFuelBenefitsControllerISpec.scala:59]: emptyCarVanFuelModel is unused and can be removed

**Review by ina-tu — CHANGES_REQUESTED** [review]
needs a rebase

File: app/controllers/employment/CompanyCarFuelBenefitsController.scala (hunk 1 above)
Comment (lz1kwn) [reply app/controllers/employment/CompanyCarFuelBenefitsController.scala:85]: removed TODO

**Review by ina-tu — APPROVED** [review]

**Pull request merged**